core/util/eventtarget.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * noVNC: HTML5 VNC client
+ * Copyright 2017 Pierre Ossman for Cendio AB
+ * Licensed under MPL 2.0 (see LICENSE.txt)
+ *
+ * See README.md for usage and integration instructions.
+ */
+
+export var EventTargetMixin = {
+    _listeners: null,
+
+   addEventListener: function(type, callback) {
+      if (!this._listeners) {
+         this._listeners = {};
```

Comment:
should just use a `Map` to avoid accidental collisions with built-in properties, etc.

Reply:
How would a `Map` improve things in that regard?

Reply:
Ah, I see what you mean now. Will fix.